corehq/apps/app_manager/models.py

```diff
@@ -5779,6 +5778,14 @@ def save(self, force_insert=False, force_update=False, using=DEFAULT_DB_ALIAS, u
             force_insert=force_insert, force_update=force_update, using=using, update_fields=update_fields
         )
 
+    @property
+    def app(self):
+        if not self._app:
+            app = get_app(self.domain, self.app_id, latest=True, target='release')
+            assert self.app_id == app.master_id, "this class doesn't handle copy app ids"
```

Comment:
If we can't quickcache `get_app` since it's hard to bust the cache, we could pull out a separate method such as `is_master_app` and quickcache it. It would be easy to bust the cache, we could avoid fetching the app on every request that way.

Reply:
It looks like more things are needed though like `is_released and version`, still possible to cache these?

Reply:
I'm not sure I follow, you're saying just cache the assertion? That's not really the issue, the app was being loaded for every request instead of only requests that don't have a cached value.